### PR TITLE
Ghost Hotel Update

### PR DESCRIPTION
**Pull request by Kush1Push1**
## About The Pull Request

Added initial code to add variety of rooms in ghost hotel (a blue ball at Ghost Cafe).

Also, removed annoying bell sound inside rooms.

<details>

https://user-images.githubusercontent.com/78963858/192285182-526492b2-62b6-4313-a1d0-49d8aa1742a4.mp4

</details>

## How This Contributes To The Skyrat Roleplay Experience

A better roleplay experience in Ghost Cafe.

## Changelog

:cl: SishTis
add: Added rooms variety to ghost cafe's hotel.
add: Added fancy cobblestone pool floor. You can reskin pool floor tile in hand to get it.
del: Silenced bell ring in ghost cafe's hotel rooms.
/:cl:

### CODE REVIEW

File: code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm [hunk 1]
```diff
@@ -7,6 +7,10 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
 	icon_state = "hilbertshotel"
 	w_class = WEIGHT_CLASS_SMALL
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	//SKYRAT EDIT ADDITION - GHOST HOTEL UPDATE
+	var/list/static/hotel_maps = list("Generic", "Apartment")
```
Comment (GoldenAlpharex) [inline code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm:11]: Add documentation to these two variables please.

File: code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm [hunk 2]
```diff
@@ -94,6 +101,11 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
 		else if(!user.dropItemToGround(src))
 			to_chat(user, span_warning("You can't seem to drop \the [src]! It must be stuck to your hand somehow! Prepare for unforeseen consequences..."))
 
+	//SKYRAT EDIT ADDITION - GHOST HOTEL UPDATE
+	var/chosen_room
+	if(istype(src, /obj/item/hilbertshotel/ghostdojo)) //to don't add another one var
+		chosen_room = tgui_input_list(user, "Choose desired room:","Time to choose", hotel_maps)
```
Comment (GoldenAlpharex) [inline code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm]: You need to handle the client not clicking any of the available options and just clicking X instead (which will return null in `chosen_room`).

File: code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm [hunk 3]
```diff
@@ -138,12 +150,18 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
 		return TRUE
 	return FALSE
 
-/obj/item/hilbertshotel/proc/sendToNewRoom(roomNumber, mob/user)
+/obj/item/hilbertshotel/proc/sendToNewRoom(roomNumber, mob/user, chosen_room) //SKYRAT EDIT ADDITION - GHOST HOTEL UPDATE
```
Comment (GoldenAlpharex) [inline code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm]: You need to mention what that change you did is, so it's not a pain to figure out when it gets changed from upstream.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I don't think that hidden room in the bottom-right corner should exist.

Furthermore, the walls you used aren't indestructible walls, which means anyone with a welder could just break into space with it. Which is very much not how it's intended.

The sink looks really awkward placed in the kitchen like that, and I also don't really know if it's really necessary in the first place.

The bathroom feels a bit weird, with the washing machine in there. I think it could use a bit more work, I'm not even sure that there's a sink and a mirror in there. You can take a bit more space for it, by removing the hidden room to its right.

The pool in the living room is a bit of a stretch, and I don't think it should be in there, it's too much.

File: code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm (hunk 3 above)
Comment (Kush1Push1) [reply code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm]: Done

File: code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm (hunk 2 above)
Comment (Kush1Push1) [reply code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm]: Done

File: code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm (hunk 1 above)
Comment (Kush1Push1) [reply code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm:11]: Good enough? (In recent commit)

**Comment by Kush1Push1** [comment]
<details>

dreamseeker_HFUgBqwnAX

</details>

**Comment by Kush1Push1** [comment]
> I don't think that hidden room in the bottom-right corner should exist.

Done.

> Furthermore, the walls you used aren't indestructible walls, which means anyone with a welder could just break into space with it. Which is very much not how it's intended.

Replaced.

> The sink looks really awkward placed in the kitchen like that, and I also don't really know if it's really necessary in the first place.

Moved to other place.

> The bathroom feels a bit weird, with the washing machine in there. I think it could use a bit more work, I'm not even sure that there's a sink and a mirror in there. You can take a bit more space for it, by removing the hidden room to its right.

It's common for me, but alright. A sink and mirror were there.

> The pool in the living room is a bit of a stretch, and I don't think it should be in there, it's too much.

Moved.

**Comment by Kush1Push1** [comment]
@GoldenAlpharex ?

<details>

image

</details>

**Comment by tf-4** [comment]
@GoldenAlpharex

File: code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm [hunk 4]
```diff
@@ -94,6 +103,11 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
 		else if(!user.dropItemToGround(src))
 			to_chat(user, span_warning("You can't seem to drop \the [src]! It must be stuck to your hand somehow! Prepare for unforeseen consequences..."))
 
+	//SKYRAT EDIT ADDITION - GHOST HOTEL UPDATE
+	var/chosen_room = "Nothing"
+	if(istype(src, /obj/item/hilbertshotel/ghostdojo)) //to don't add another one var
+		chosen_room = tgui_input_list(user, "Choose desired room:","Time to choose", hotel_maps)
```
Comment (GoldenAlpharex) [inline code/modules/mapfluff/ruins/spaceruin_code/hilbertshotel.dm]:
```suggestion
		chosen_room = tgui_input_list(user, "Choose desired room:", "Time to choose", hotel_maps)
```

File: modular_skyrat/master_files/code/game/turfs/closed/_closed.dm [hunk 5]
```diff
@@ -0,0 +1,7 @@
+/turf/closed/indestructible/steel
```
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/game/turfs/closed/_closed.dm:1]: I'd rename this to this, for less confusion (yes you'll have to update it in your map too, sorry!
```suggestion
/turf/closed/indestructible/iron
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
The wooden decals on the carpet look a bit off, because they're not uniform on any side and contrast heavily with the carpet's very defined edges.

I'm still entirely sold on the pool in the bathroom, unless you're telling me it's supposed to be a bath. In which case, it's a bit big for a bath, but I can see the reasoning. I'd just shrink it down a bit to make it more obvious.

**Comment by Kush1Push1** [comment]
> The wooden decals on the carpet look a bit off, because they're not uniform on any side and contrast heavily with the carpet's very defined edges.
> 
> I'm still entirely sold on the pool in the bathroom, unless you're telling me it's supposed to be a bath. In which case, it's a bit big for a bath, but I can see the reasoning. I'd just shrink it down a bit to make it more obvious.

I can't use "iron" at the end, it will override other code.

<details>

Code_PH9P8LLimq

dreamseeker_cUYCvQSgo0

</details>

I don't think I have enough will to work on it any further. I moved the bath and filled up unused space with a placeholder.

**Comment by Kush1Push1** [comment]
Oof -.-

**Comment by Wildwestmars** [comment]
My friend and I just got done playtesting the new apartments and...
The bathroom floods the entire complex as soon you open the door. There is also a floating painting frame in the living room.
Besides that, it was quite fun (and cozy)!

Note: My friend is quite excited for the pool/Japanese style bath to work!

**Comment by Kush1Push1** [comment]
> My friend and I just got done playtesting the new apartments and... The bathroom floods the entire complex as soon you open the door. There is also a floating painting frame in the living room. Besides that, it was quite fun (and cozy)!
> 
> Note: My friend is quite excited for the pool/Japanese style bath to work!

Strange, I'll check it 🤔

**Comment by Kush1Push1** [comment]
1) Fixed the map. The pool doesn't spill its water now and the paint was moved back on the wall (there was an issue with its direction).

2) Added fancy subtype of pool, with look like cobblestone and made it craftable via reskin of pool floor tile.

**Comment by Kush1Push1** [comment]
<details>

The pool
dreamseeker_juE5EomunT

The paint
dreamseeker_Lf9nigeuTY

</details>

**Comment by theselfish** [comment]
The bathroom needs a drain, the water spills and floods the entire room otherwise.
 Look at how the Cafe's pool does it for an example.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Apparently the bathroom needs a drain. Aside from that, it's all good with me, I'd say.

**Review by GoldenAlpharex — APPROVED** [review]

**Comment by GoldenAlpharex** [comment]
I tried it myself and apparently that wasn't an issue, the only thing I've noticed is the fishes dying, but that's about it. It's good to send.

**Pull request merged**